### ISSUE TITLE
Version manipulation 

### ISSUE THREAD

**Issue by Vishnu-2810** [anthonyharrison/lib4sbom#32]
As per semantic versioning standards , pre release versions can contain hyphens but in lib4sbom we are changing the version by splitting the hyphens . Do we have any logical reason behind this like only major versions to be detected @anthonyharrison ?

**Manipulation code**

 def _semantic_version(self, version):
        return version.split("-")[0] if "-" in version else version

Thanks in advance.

**Comment by anthonyharrison** [comment]
@Vishnu-2810 Component versions are very inconsistent. The approach was to enable vulnerability scanners to find the component in a vulnerability database. Including the pre-release component in the version component meant that potential vulnerabilities could be missed. Do you have an example of where the removal of the pre-release component has resulted in some incorrect results from a vulnerability scanner?

**Comment by Vishnu-2810** [comment]
We will also have pre-release versions vulnerabilities in the vulnerability database . Then how potential vulnerabilties could be missed . Can you please explain in detail ? 

Thanks in advance

> @Vishnu-2810 Component versions are very inconsistent. The approach was to enable vulnerability scanners to find the component in a vulnerability database. Including the pre-release component in the version component meant that potential vulnerabilities could be missed. Do you have an example of where the removal of the pre-release component has resulted in some incorrect results from a vulnerability scanner?

We will also have pre-release versions vulnerabilities in the vulnerability database . Then how potential vulnerabilties could be missed . Can you please explain in detail @anthonyharrison  ? 

Thanks in advance

**Comment by Vishnu-2810** [comment]
> @Vishnu-2810 Component versions are very inconsistent. The approach was to enable vulnerability scanners to find the component in a vulnerability database. Including the pre-release component in the version component meant that potential vulnerabilities could be missed. Do you have an example of where the removal of the pre-release component has resulted in some incorrect results from a vulnerability scanner?

We will also have pre-release versions vulnerabilities in the vulnerability database . Then how potential vulnerabilties could be missed . Can you please explain in detail @anthonyharrison ? 

Thanks in advance

**Comment by ValdikSS** [comment]
>Do you have an example of where the removal of the pre-release component has resulted in some incorrect results from a vulnerability scanner?

Yes. Take a look at GhostScript Debian package: https://packages.debian.org/bookworm/ghostscript
The current version is `10.0.0~dfsg-11+deb12u4`, where [several CVEs were patched](https://metadata.ftp-master.debian.org/changelogs//main/g/ghostscript/ghostscript_10.0.0~dfsg-11+deb12u4_changelog):

```
ghostscript (10.0.0~dfsg-11+deb12u4) bookworm-security; urgency=high

  * Non-maintainer upload by the Security Team.
  * In SAFER (default) don't allow eexec seeds other than the Type 1 standard
    (CVE-2023-52722)
  * Uniprint device - prevent string configuration changes when SAFER
    (CVE-2024-29510)
  * Bug #707691 (CVE-2024-33869)
  * Bug 707691 part 2 (CVE-2024-33869)
  * Bug #707686 (CVE-2024-33870)
  * OPVP device - prevent unsafe parameter change with SAFER (CVE-2024-33871)
```

The previous, vulnerable version is `10.0.0~dfsg-11+deb12u3`. [Snyk database](https://security.snyk.io/package/linux/debian:12/ghostscript) has data about this exact version.

![Screenshot_20240720_201541-fs8](https://github.com/user-attachments/assets/7dfd4dc8-ddc5-4696-8076-95706dc72f06)

distro2SBOM strips the version down to `10.0.0~dfsg`, which doesn't allow to track vulnerabilities.
https://github.com/anthonyharrison/distro2SBOM/issues/19

**Comment by anthonyharrison** [comment]
@ValdikSS The latest working version of Lib4sbom has removed the _sematic_version check. Can you try this with distro2sbom and see if this removes your problem?

**Comment by ValdikSS** [comment]
> @ValdikSS The latest working version of Lib4sbom has removed the _sematic_version check. Can you try this with distro2sbom and see if this removes your problem?

Which one? I've tested on 0.7.1.

**Comment by anthonyharrison** [comment]
You need to use the latest version in the GitHub repository. It will become
0.7.2.

**Comment by ValdikSS** [comment]
Yes, looks good.

**Comment by anthonyharrison** [comment]
Fixed in version 0.7.2
